src/decoding/extrinsic_type_info.rs

```diff
@@ -97,23 +108,32 @@ impl<'a> core::fmt::Display for ExtrinsicInfoError<'a> {
                 pallet_index,
                 pallet_name,
             } => {
-                write!(f, "Cannot find calls type with id {id} in pallet '{pallet_name}' (pallet index {pallet_index})")
+                write!(f, "Cannot find calls type with id {id} in pallet '{pallet_name}' (pallet index {pallet_index}).")
             }
             ExtrinsicInfoError::CallsTypeShouldBeVariant {
                 id,
                 pallet_index,
                 pallet_name,
             } => {
-                write!(f, "Calls type with id {id} should be a variant in pallet '{pallet_name}' (pallet index {pallet_index})")
+                write!(f, "Calls type with id {id} should be a variant in pallet '{pallet_name}' (pallet index {pallet_index}).")
             }
             ExtrinsicInfoError::ExtrinsicTypeNotFound { id } => {
-                write!(f, "Could not find the extrinsic type with id {id}")
+                write!(f, "Could not find the extrinsic type with id {id}.")
             }
             ExtrinsicInfoError::ExtrinsicAddressTypeNotFound => {
-                write!(f, "Could not find the extrinsic address type")
+                write!(f, "Could not find the extrinsic address type.")
             }
             ExtrinsicInfoError::ExtrinsicSignatureTypeNotFound => {
-                write!(f, "Could not find the extrinsic signature type")
+                write!(f, "Could not find the extrinsic signature type.")
+            }
+            ExtrinsicInfoError::ExtrinsicExtensionVersionNotSupported { extension_version } => {
+                // Dev note: If we see a V5 GEneral extrinsic, it will have a byte for the version of the transaction extensions.
+                // In V15 or below metadata, we don't know which set of transaction extensions we're being told about though. Is it
+                // the set that corresponds to the version byte we see or not?
+                write!(f, "The extrinsic contains an extension version (here, {extension_version}), but in metadata <=V15 it's not clear if we can decode this version or not.")
```

Comment:
I think it would useful with some info such as `Note: Metadata <=V15 only supports transaction status version 0` 

I find `but in metadata <=V15 it's not clear if we can decode this version or not` a bit tricky to understand as user

Reply:
Metadata V15 doesn't necessarily only support version 0 though. The actual issue is that in V15 metadata we are told about the current latest version of signed extensions, but we don't know what that version is.

So right now, there is only one version (0) so it's all good. When the code allows for more than one version, we'll actually have to disallow _any_ version from decoding via V15 and below metadata becasue we don't know if the version aligns with what metadata is telling us or not.

(that's why it's ahrd to explain; more tricky than just "0 works in <V16" :D)